compression/lpf.py

```diff
@@ -0,0 +1,73 @@
+
+from string_indexing import lcp, suffix_array
+
+
+def naive(w, n):
+  ret = []
+  for i in range(1, n + 1):
+    right = w[i:]
+    while True:
+      if right in w[1:(i + len(right) - 1)]:
+        ret += [len(right)]
+        break
+      right = right[:-1]
+  return [-1] + ret
+
+
+def chrochemore_ille_smith(text, n, SA=None, LCP=None):
```

Comment:
Overall I think that you can merge functions (crochemore... and its respective compute_cis...) + imitate behavior from lcp.from_suffix array, that is just require LCP and SA as parameters, and not to compute them.

Reply:
Since you suggested previously to keep the authors in the functions name, should it be `crochemore_ilie_smyth_no_stack_from_array`? Isn't that too long?